### ISSUE TITLE
Cutscenes

### ISSUE THREAD

**Issue by birkirarndal** [haframjolk/the-quest-game#18]

**Comment by haframjolk** [comment]
Initial work on cutscenes started

**Comment by haframjolk** [comment]
Implemented slave character system today as well as having Raudalaekur's cutscene transition more seamlessly into the player's correct state.

**Comment by haframjolk** [comment]
Cutscenes will not be ready in time for verkefni 4 (releasing today), moved milestone to verkefni 5

**Comment by haframjolk** [comment]
Doorbell sound and dramatic sound effect ([dun-dun-duuuuuun](https://www.youtube.com/watch?v=cphNpqKpKc4)) needed for Raudalaekur cutscene (@birkirarndal)

**Comment by haframjolk** [comment]
Svennz's house cutscene now includes doorbell sound. Raudalaekur opening cutscene now has Svennz walk out of frame in the end and includes sound effects. Consider adding zoom effect to dramatic sound part?

**Comment by haframjolk** [comment]
Added interactable objects to Raudalaekur scene (required to progress) and did touchups.

**Comment by haframjolk** [comment]
Added cutscenes to New Zealand, implemented Herbert & Kiwi NPC functionality required. Made a lot of progress today.

**Comment by haframjolk** [comment]
Logo added, very convoluted Rasmusen chase scene system with signal and function calls all over the place implemented for Raudalaekur.

**Comment by haframjolk** [comment]
Today: Rebased cutscene on develop and fixed merge conflicts. Did some touchups on Raudalaekur and created a Rasmusen chase scene for NewZealand. Connected that one to YonasCastle, added guard & Yonas prefabs and created a cutscene for the guard. To do: investigate why collecting sound does not play in NZ cutscene, consider adding camera zoom to dramatic SFX in Raudalaekur, get battle system working and implement in Yonas cutscene, also add henchman battle scene in Raudalaekur which sends player to NZ. Edit: also consider changing logo background.

**Comment by haframjolk** [comment]
Today pt. 1: Rebased cutscene on develop to get battle system, connected battle system to cutscenes and adjusted as needed. Set up battle in Raudalaekur and added fadeouts to scene changes. Changed some dialogue and increased walking speed in older cutscenes. NZ sound issue workaround applied by using signal instead of playing audio in timeline. Still to do: look into camera zoom in Raudalaekur, replace prefabs in battle system, add battle to Yonas's castle, create credits scene.

**Comment by haframjolk** [comment]
Today pt. 2: Added camera zoom, replaced prefabs in battle system, added battle to Yonas's castle, created "Item Get" display and added to cutscenes, created credits scene. Still left to do: change battle stats of players and enemies so that losing is possible.

**Comment by haframjolk** [comment]
Today pt. 1: Added "Item Get" display to stone in Raudalaekur, created pause menu. To do: same as yesterday, plus add "Item Get" and "Item Give" for kiwi scenes in NZ, consider adding fade in to scenes where fade-out is placed in between. Consider shortening wait time before credits start rolling?

**Comment by haframjolk** [comment]
Today pt. 2: Added "Item Get" and "Item Give" to kiwi scenes in NZ, still waiting for kiwi graphics as per #41. Fine-tuned timing of multiple fades as well as time between loading scene and the cutscene starting. Changed the game logo background in Raudalaekur and did some organization. Added myself to graphics credits as I did work on the logo and the general presentation of everything. To do: replace placeholder item texture in NZ "Item Get" and "Item Give" canvases with kiwifruit when graphic is ready. Otherwise, the branch is ready to merge and we can finally move on to our remaining issues (battle stats moved to separate issue).
